From b922515e0761295bdacfc1f3fdd113ce4f231ff3 Mon Sep 17 00:00:00 2001
From: Xogium <contact@xogium.me>
Date: Thu, 14 Oct 2021 09:32:40 +0200
Subject: [PATCH 07/10] stm32mp157c-odyssey: enable usb ports.

This commit enables both type a ports on the odyssey, in ehci and ohci mode, as well as the usb-C port for OTG.
---
 arch/arm/boot/dts/stm32mp157c-odyssey.dts | 39 +++++++++++++++++++++++
 1 file changed, 39 insertions(+)

diff --git a/arch/arm/boot/dts/stm32mp157c-odyssey.dts b/arch/arm/boot/dts/stm32mp157c-odyssey.dts
index 37a8402ea93b..1bd6d6f03acc 100644
--- a/arch/arm/boot/dts/stm32mp157c-odyssey.dts
+++ b/arch/arm/boot/dts/stm32mp157c-odyssey.dts
@@ -136,3 +136,42 @@ &i2c4 {
 &dts {
 	status = "okay";
 };
+
+&usbphyc {
+	status = "okay";
+};
+
+&usbh_ehci {
+	phys = <&usbphyc_port0>, <&usbphyc_port1 1>;
+	phy-names = "usb1-phy", "usb2-phy";
+	status = "okay";
+};
+
+&usbh_ohci {
+	phys = <&usbphyc_port0>, <&usbphyc_port1 1>;
+	phy-names = "usb1-phy", "usb2-phy";
+	status = "okay";
+};
+
+&usbphyc_port0 {
+	phy-supply = <&vdd_usb>;
+	vbus-supply = <&usb1vbus>;
+	vdda1v1-supply = <&reg11>;
+	vdda1v8-supply = <&reg18>;
+};
+
+&usbphyc_port1 {
+	phy-supply = <&vdd_usb>;
+	vbus-supply = <&usb2vbus>;
+	vdda1v1-supply = <&reg11>;
+	vdda1v8-supply = <&reg18>;
+};
+
+&usbotg_hs {
+	compatible = "st,stm32mp15-fsotg", "snps,dwc2";
+	pinctrl-names = "default";
+	pinctrl-0 = <&usbotg_hs_pins_a &usbotg_fs_dp_dm_pins_a>;          /* configure OTG ID and full-speed data pins */
+	vbus-supply = <&vbus_otg>;                                        /* voltage regulator to supply Vbus */
+	dr_mode = "peripheral";
+	status = "okay";
+};
-- 
2.33.0

